packages/babel-generator/src/printer.ts

```diff
@@ -241,29 +257,61 @@ class Printer {
    * Add a newline (or many newlines), maintaining formatting.
    */
 
-  newline(i: number = 1): void {
-    if (this.format.retainLines || this.format.compact) return;
+  newline(i: number = 1, force?: boolean): void {
+    this.newlineRelative(i, force);
 
-    if (this.format.concise) {
-      this.space();
-      return;
-    }
+    // if (this.format.retainLines || this.format.compact) return;
```

Comment:
Is this code not needed anymore? If so, can yo write in a comment why are we commenting it out rather than just deleting it?

It seems like `newline` is just an alias for `newlineRelative`, why don't we only use one of them?

Reply:
At first I wanted to make `newlineRelative` a new method, which would be clearer, and make `newline` another method with nuances, then switch most of the original usage to `newlineRelative`. Later it seemed unnecessary, and now I removed `newline` and renamed `newlineRelative` to that.